### ISSUE TITLE
Compiling hangs when selectors with ">" are present

### ISSUE THREAD

**Issue by massimo-cassandro** [sass/libsass#3013]
Compiling scss files hangs in presence of selectors that contain the ">" character. (version 3.6.2)

Example:

this rules hang while compiling:

```
// 1
.buttons > div {
  //some rules

  > button { 
    //some rules
  }
}

// 2
label {
  > div {
    // some rules

    &:nth-child(1) {
      // some rules
    }
  }
}
```

All works well if rewritten in this way:
```
// 1
.buttons > div {
  //some rules
}

.buttons > div > button { 
  //some rules
 }

// 2
label {
  > div {
    // some rules
  }
}
label > div:nth-child(1) {
     // some rules
}
```

Expected css:

```
// 1
.buttons > div { ... }
.buttons > div > button { ... }

// 2
label > div { ... }
label > div:nth-child(1) { ... }
```

**Comment by bdkjones** [comment]
I can’t confirm the exact cause, but I can confirm that I’ve had a flood of “sass compiling just hangs” reports since I shipped 3.6.2. I am rolling back to 3.6.1 for the time being.

**Comment by mgreter** [comment]
This seems to be the same issue as already reported.
Looks like already fixed either by #3007 or #2995.

**Comment by mgreter** [comment]
Sorry for the inconvenience @bdkjones, too bad the specs did not catch it before.
Will probably release 3.6.3 in some days if nothing else critical is reported.

**Comment by mgreter** [comment]
Please re-open if you can reproduce this with 3.6.2, thanks.

**Comment by bdkjones** [comment]
@mgreter no worries; I ship bugs all the time. It happens. This report is from one of my users and he was definitely running 3.6.2 at the time. I didn’t try to reproduce it myself, but I did have many other reports of hung compiling with 3.6.2, so I just assumed there was a regression somewhere.

All of my Sass test files compiled correctly before I shipped the update with Libsass 3.6.2 in it, so the hang only manifests in certain conditions. I’ll wait for 3.6.3 and see if that resolves it.

@massimo-cassandro: can you please email me the original Sass files that were hanging the compiler? I need ALL of them—the main file and everything it imports. That way, I can test Libsass 3.6.3 when it’s ready. My existing Sass test files clearly don’t trigger this bug.
